tests/test_visualization.py

```diff
@@ -39,3 +40,46 @@ def fail_objective(_):
 
     study.optimize(fail_objective, n_trials=1)
     assert len(data) == 1
+
+
+def test_get_optimization_history_plot():
+    # type: () -> None
+
+    study = create_study()
+
+    # Test with no trial.
+    figure = _get_optimization_history_plot(study)
+    assert len(figure.data[0].x) == 0
+    assert len(figure.data[0].y) == 0
+    assert len(figure.data[1].x) == 0
+    assert len(figure.data[1].y) == 0
+
+    def objective(trial, n_iterate):
+        # type: (Trial, int) -> float
+
+        if n_iterate == 0:
+            return 1.0
+        elif n_iterate == 1:
+            return 2.0
+        elif n_iterate == 2:
+            return 0.0
+        return 0.0
+
+    # Test with a trial
+    for i in range(3):
+        study.optimize(lambda t: objective(t, i), n_trials=1)
+    figure = _get_optimization_history_plot(study)
+    assert len(figure.data) == 2
+    assert tuple(figure.data[0].x) == (0, 1, 2)
+    assert tuple(figure.data[0].y) == (1.0, 2.0, 0.0)
+    assert tuple(figure.data[1].x) == (0, 1, 2)
+    assert tuple(figure.data[1].y) == (1.0, 1.0, 0.0)
+
+    # Ignore failed trials.
+    def fail_objective(_):
+        # type: (Trial) -> float
+
+        raise ValueError
+
+    study.optimize(fail_objective, n_trials=1)
```

Comment:
It seems that re-evaluation of `figure` (i.e., `figure = _get_optimization_history_plot(study)`) is needed after this optimization.